smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/providers/HttpHeaderProvider.kt

```diff
@@ -66,8 +66,14 @@ class HttpHeaderProvider(
     }
 
     fun renderProvider(writer: SwiftWriter) {
-        writer.openBlock("extension \$N: \$N {", "}", inputSymbol, ClientRuntimeTypes.Middleware.Providers.HeaderProvider) {
-            writer.openBlock("public var headers: \$N {", "}", ClientRuntimeTypes.Http.Headers) {
+        writer.openBlock("extension \$N {", "}", inputSymbol) {
+            writer.write("")
+            writer.openBlock(
+                "static func headerProvider(_ value: \$N) -> \$N {",
+                "}",
+                inputSymbol,
+                ClientRuntimeTypes.Http.Headers,
+            ) {
```

Comment:
- Removed protocol conformance from the extension
- Extension declares a static closure instead of a protocol property
- Kotlin cleanup